### DISCUSSION TITLE
useFormContext + useFieldArray fields have not changed

### DISCUSSION THREAD

**Discussion by SturdyCat** [react-hook-form/react-hook-form#8229] [General]
### Version Number

7.18.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-v7-bug-06vmyz

### Steps to reproduce

1.  Click on 'click append‘ button
2. See error
![image](https://user-images.githubusercontent.com/16874676/164387900-4e021e2f-ea77-4bbb-a951-ea82eb022434.png)

### Expected behaviour

useFieldArray fields synchronize changes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
hi there, the `name` is unique we don't support multiple useFieldArray with the same name. there is a feature PR, which may help your use case: https://github.com/react-hook-form/react-hook-form/pull/7544

here is a codesandbox with the above PR: https://codesandbox.io/s/cocky-franklin-sn0vn?file=/src/App.js:1373-1383

**Comment by isnifer** [comment]
I think the suggestion even wrong, lines (13-17):
```js
const { control } = useFormContext()
const { fields } = useFieldArray({
    control, // control props comes from useForm (optional: if you are using FormContext)
    name: "data" // unique name for your Field Array
});
```

should be replaced with:
```js
const { watch } = useFormContext()
const fields = watch('data')
```

And then everything works fine:
<img width="1103" alt="image" src="https://user-images.githubusercontent.com/1788245/164523187-6900523a-d1f7-4bd4-a3ef-1d7e6e6e7992.png">

**Comment by Moshyfawn** [comment]
If the OP just wants to `watch` the `fields`'s values, then, yea, sure, you can simply use the `watch` method instead.

The OG CSB includes `useFieldArray` hook call to get the `fields`, which might mean, that they want to also manipulate the field array data in multiple places. This behavior is not supported by RHF at this time, hence [the suggestion with the CSB](https://github.com/react-hook-form/react-hook-form/issues/8227#issuecomment-1104800409). The provided CSB contains the unreleased RHF bundle with the proposed field array `Context` changes, which would solve the OP's issue

**Comment by bluebill1049** [comment]
yes, the watch works for value render, I think if the user does want to sync field array. the FormContext with field array is a good utility to be included which is currently in PR.

**Comment by Moshyfawn** [comment]
I'll give the feature another thought, 'cuz I don't exactly like how it clutters the component body, but we might wanna just move ahead with it

**Reply by bluebill1049** [reply]
I think having form context is just a convenient syntax (with a bit performance sacrifc) ofc users should always measure against their app, they can do the same with props anyhow, but keen to hear everyone's thoughts on that PR. as it only changes with type without adding more bytes.

**Comment by Moshyfawn** [comment]
Maybe move the issue to discussion?

**Comment by bluebill1049** [comment]
yea move to discussion.
